### PR TITLE
FS-505 fund validation

### PR DESCRIPTION
**Pull request by SentientHousePlant**
If eligible than you are redirected to current xgov form runner with our baseline questions.

Otherwise you are taken to a not eligible page.

### CODE REVIEW

**Comment by SentientHousePlant** [comment]
Screenshot 2022-02-11 at 10 34 03

File: app/forms/eligibility_questions.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+        money_field = IntegerField(
+            label="project_money_amount",
+            validators=[DataRequired(), NumberRange(max=max_amount)],
+        )
+
+    def min_money_page():
+
+        form = MinimiumMoneyForm()
+        if form.money_field.data is not None and not form.validate_on_submit():
```
Comment (harryyo) [inline app/forms/eligibility_questions.py:34]: Could `if not form.money_field.data` be a pythonic alternative?

File: app/create_app.py [hunk 2]
```diff
@@ -10,11 +11,14 @@
...
-    csrf.init_app(flask_app)
-
     flask_app.config.from_pyfile("config.py")
 
+    # Auto build the static assets if we are in the dev environment.
+    # In production we will most likely install static assets differently.
+    if flask_app.config.get("ENV") == "development":
+
+        build_assets()
+
```
Comment (harryyo) [inline app/create_app.py:21]: This looks like it will run on every request. Perhaps should go somewhere else..?

Comment (SentientHousePlant) [reply app/create_app.py:21]: Any suggestions for this? Would be nice to have very little dev work required.

File: app/forms/eligibility_questions.py (hunk 1 above)
Comment (SentientHousePlant) [reply app/forms/eligibility_questions.py:34]: Tried this and it didn't work. Will refactor this today to be more pythonic.

File: app/forms/eligibility_questions.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    max_amount: int, success_url: str, data_hook: callable = print
+):
+    """Returns a Flask view function which checks the eligibility
+    of a fund using the gov.uk design system.
+
+    Args:
+        max_amount (int): The maximium amount one can apply for.
+        success_url (str): The url one is redirected to if eliglible.
+        data_hook (callable): This function will be ran on the inputted
+        data if the fund is eligible. By default prints to console.
```
Comment (ramsharma-prog) [inline app/forms/eligibility_questions.py:23]: Nice docstrings!

File: app/templates/eligible.html [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{% extends "base.html" %}
+
+{%- from 'govuk_frontend_jinja/components/input/macro.html' import govukInput -%}
```
Comment (ramsharma-prog) [inline app/templates/eligible.html:3]: Do we need this line?

File: app/templates/eligible.html [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    <p  class="govuk-body-l">
+      Based on your answers, you are eligible to apply for this fund.
+    </p>
+</div>
+
+{{ govukButton({
+    "isStartButton": True,
+    "href" : service_url,
+  "text": "Start now"
+}) }}
```
Comment (ramsharma-prog) [inline app/templates/eligible.html:20]: Where do you find these button options?

File: app/templates/min_funding_amount.html [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+<form method="POST">
+  {{ form.csrf_token }}
+{{
+  govukInput({
+    "label": {"text": "How much money do you need?"},
+    "name": form.money_field.id,
+    "label": {"text": "Enter your answer using numbers only"},
+    "prefix": {"text": "£"},
+    "classes": "govuk-input--width-20",
+    "type" : "number"
```
Comment (ramsharma-prog) [inline app/templates/min_funding_amount.html:21]: Where do we find Input options?

**Review by ramsharma-prog — CHANGES_REQUESTED** [review]

File: app/templates/eligible.html (hunk 4 above)
Comment (SentientHousePlant) [reply app/templates/eligible.html:3]: The components and their options can be found here https://govuk-frontend-jinja.herokuapp.com

File: app/templates/min_funding_amount.html (hunk 6 above)
Comment (SentientHousePlant) [reply app/templates/min_funding_amount.html:21]: The components and their options can be found here https://govuk-frontend-jinja.herokuapp.com

File: app/templates/eligible.html (hunk 5 above)
Comment (SentientHousePlant) [reply app/templates/eligible.html:20]: The components and their options can be found here https://govuk-frontend-jinja.herokuapp.com

**Review by ramsharma-prog — APPROVED** [review]

File: app/forms/eligibility_questions.py (hunk 1 above)
Comment (harryyo) [reply app/forms/eligibility_questions.py:40]: How about only validating the form if the `if request.POST and form.validate_on_submit():`

Comment (SentientHousePlant) [reply app/forms/eligibility_questions.py]: Hi Harry, 
Thanks for the insightful suggestion. I've added this change :)
Please re-review and let me know if you're happy with this PR.
J

**Review by harryyo — APPROVED** [review]

**Pull request merged**